recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,218 @@
-from conans import ConanFile, tools
-from conans.tools import Version
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import copy, get, rmdir
+from conan.tools.scm import Version
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.microsoft import check_min_vs, is_msvc
+
 import os
 
+required_conan_version = ">=1.52.0"
+
 class Stlab(ConanFile):
     name = 'stlab'
     description = 'The Software Technology Lab libraries.'
     url = 'https://github.com/conan-io/conan-center-index'
     homepage = 'https://github.com/stlab/libraries'
     license = 'BSL-1.0'
-    topics = 'conan', 'c++', 'concurrency', 'futures', 'channels'
+    topics = 'concurrency', 'futures', 'channels'
 
-    settings = "arch", "os", "compiler", "build_type", 
+    settings = "arch", "os", "compiler", "build_type",
 
     options = {
-        "boost_optional": [True, False],
-        "boost_variant": [True, False],
-        "coroutines": [True, False],
+        "with_boost": [True, False],
+        "no_std_coroutines": [True, False],
+        "future_coroutines": [True, False],
```

Comment:
This does not impact the build or package contents, https://github.com/conan-io/conan-center-index/blob/20829d05ef9da4339c1a9e806ed2f91e53a71126/recipes/stlab/all/conanfile.py#L208

Maybe it can be deleted or just left to the consumer?

Reply:
Why the consumer has to know about the internals of the project?

Reply:
They have to know either way what this option does? It should be deleted from the package ID if you want to keep it :)

Reply:
Yes, the user has to know about it, but this is why there is an option for it. 
I do not agree that the user has to set the CMake macros/defines/envvars